src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```diff
@@ -107,10 +115,23 @@ public ReadsSparkSource(JavaSparkContext ctx) {
     /**
      * Loads the header using Hadoop-BAM.
      * @param filePath path to the bam.
+     * @param auth authentication information if using GCS.
      * @return the header for the bam.
      */
-    public static SAMFileHeader getHeader(final JavaSparkContext ctx, final String filePath) {
+    public static SAMFileHeader getHeader(final JavaSparkContext ctx, final String filePath, final AuthHolder auth) {
```

Comment:
I'd rather not. Overloads are a maintenance burden and here they'd send the wrong message. Here we *want* our code to pass the AuthHolder object rather than assuming that the user-provided path is not on GCS. Therefore we shouldn't be making it easier to write code that'll break on GCS.